pkg/slayers/path/scion/raw_spec.gobra

```diff
@@ -232,31 +232,21 @@ pure func hopFields(
 }
 
 ghost
-requires -1 <= currHfIdx && currHfIdx < len(hopfields)
-ensures len(res) == currHfIdx + 1
-decreases currHfIdx + 1
-pure func segPast(hopfields seq[io.IO_HF], currHfIdx int) (res seq[io.IO_HF]) {
-	return currHfIdx == -1  ?
-		seq[io.IO_HF]{} :
-		seq[io.IO_HF]{hopfields[currHfIdx]} ++ segPast(hopfields, currHfIdx - 1)
+ensures len(res) == len(hopfields)
+decreases len(hopfields)
+pure func segPast(hopfields seq[io.IO_HF]) (res seq[io.IO_HF]) {
+	return len(hopfields) == 0 ? seq[io.IO_HF]{} :
+		seq[io.IO_HF]{hopfields[len(hopfields)-1]} ++ segPast(
+			hopfields[:len(hopfields)-1])
```

Comment:
```suggestion
		seq[io.IO_HF]{hopfields[len(hopfields)-1]} ++ segPast(hopfields[:len(hopfields)-1])
```

Reply:
There's an interesting feature in Gobra: the ++ operator cannot be at the beginning or the end of a line. Given that the entire line is quite long, this is the best place to split it.